### DISCUSSION TITLE
Any way to automatically Generate PDF Previews

### DISCUSSION THREAD

**Discussion by dave2084** [awcodes/filament-curator#525] [General]
I'm in the process of switching my models from Spatie media Library to Curator but I have a lot of PDFs.  Currently when I upload or seed a PDF I use conversions to render JPG of the PDF first page for a preview.  

Is there any way to do this in Curator other than doing it manually and uploading it as a curation?

**Comment by awcodes** [comment]
You could probably extend the MediaObserver to automate the process when creating or updating a record.

**Comment by dave2084** [comment]
Thanks, it looks like it will work from a quick bit of DB manipulation.

However, the Curations tab won't appear on the media edit page as it fails the check for Awcodes\Curator::is_media_resizable, that is fine though I can use spatie/pdf-to-image to generate the curations and add them to the record for this media item on upload or change then display the image on my frontend pages if the curation exists.
